core/server/src/main/java/alluxio/worker/block/DefaultBlockWorker.java

```diff
@@ -137,12 +137,34 @@ public DefaultBlockWorker(BlockMasterClient blockMasterClient,
     mMetricsReporter = new BlockMetricsReporter(WorkerContext.getWorkerSource());
     mSessions = sessions;
     mBlockStore = blockStore;
+    mWorkerId = workerId;
 
     mBlockStore.registerBlockStoreEventListener(mHeartbeatReporter);
     mBlockStore.registerBlockStoreEventListener(mMetricsReporter);
   }
 
   @Override
+  public BlockStore getBlockStore() {
+    return mBlockStore;
+  }
+
+  @Override
+  public BlockWorkerClientServiceHandler getWorkerServiceHandler() {
+    return new BlockWorkerClientServiceHandler(this);
+  }
```

Comment:
I don't think we often create new block workers, so caching wouldn't get us much performance here. I just moved this and the method above it down because they were previously placed before the constructor, but constructors should come before methods.